### PR TITLE
[MO] compress_to_fp16=True by default (2nd attempt)

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
 - this PR enables compression to fp16 by default when IR is generated from cli.

There are 4 scenarios to generate IR:
- tools.**mo.convert_model** (legacy) - no compression by default
-  **mo cli tool** (legacy) - compress_to_fp16=True by default
- openvino.**runtime.save_model** - already compress_to_fp16 by default
- **ovc cli tool** - compress_to_fp16=True by default
- tools.**ovc.convert_model** - does not have compression parameter and does not compress at al

### Tickets:
 - [CVS-115680](https://jira.devtools.intel.com/browse/CVS-115680)

### CODE REVIEW

File: tools/mo/openvino/tools/mo/convert.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def convert_model(
         source_layout: [str, Layout, dict] = (),
         target_layout: [str, Layout, dict] = (),
         layout: [str, Layout, LayoutMap, list, dict] = (),
-        compress_to_fp16: bool = False,
+        compress_to_fp16: bool = True,
```
Comment (slyalin) [inline tools/mo/openvino/tools/mo/convert.py:37]: Don't do that. We have `save_model` that is the place where we should apply fp16 compression.

File: tools/ovc/openvino/tools/ovc/cli_parser.py [hunk 2]
```diff
@@ -646,7 +646,7 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
     # Command line tool specific params
     common_group.add_argument('--output_model',
                               help='This parameter is used to name output .xml/.bin files with converted model.')
-    common_group.add_argument('--compress_to_fp16', action='store_true',
+    common_group.add_argument('--compress_to_fp16', action='store_true', default=True,
```
Comment (slyalin) [inline tools/ovc/openvino/tools/ovc/cli_parser.py:649]: Keep this parameter in `mo` CLI tool only. Not propagate to Python API `convert_model`.

Comment (pavel-esir) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:649]: It is kept in cli tool only.  If user will try to do ovc.convert(..., compress_to_fp16=True/False) he/she will end up having unexpected keyword argument error

File: tools/mo/openvino/tools/mo/convert.py (hunk 1 above)
Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/convert.py]: Agreed and reverted this line back

File: tools/ovc/openvino/tools/ovc/cli_parser.py (hunk 2 above)
Comment (popovaan) [inline tools/ovc/openvino/tools/ovc/cli_parser.py:649]: Does it work for **ovc** cli tool:
ovc --input_model INPUT_MODEL --compress_to_fp16=False

Comment (pavel-esir) [reply tools/ovc/openvino/tools/ovc/cli_parser.py]: when replaced `action='store_true'` with `type=check_bool` 
```
ovc --input_model INPUT_MODEL --compress_to_fp16=False/True
```
started to work

**Comment by popovaan** [comment]
Please move compression step from convert_model pipeline
https://github.com/openvinotoolkit/openvino/blob/8d5a0b1d53ea6d5668ac025826b7f179c866fb90/tools/ovc/openvino/tools/ovc/moc_frontend/moc_emit_ir.py#L29-L32

to ovc main 
https://github.com/openvinotoolkit/openvino/blob/8d5a0b1d53ea6d5668ac025826b7f179c866fb90/tools/ovc/openvino/tools/ovc/main.py#L28

**Comment by pavel-esir** [comment]
@popovaan done. Please review

**Review by popovaan — APPROVED** [review]

File: tools/ovc/openvino/tools/ovc/cli_parser.py [hunk 3]
```diff
@@ -646,7 +646,7 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
     # Command line tool specific params
     common_group.add_argument('--output_model',
                               help='This parameter is used to name output .xml/.bin files with converted model.')
-    common_group.add_argument('--compress_to_fp16', action='store_true',
+    common_group.add_argument('--compress_to_fp16', type=check_bool, default=True,
```
Comment (slyalin) [inline tools/ovc/openvino/tools/ovc/cli_parser.py:649]: @popovaan, @pavel-esir, how is it going to be printed in the help for ovc tool?

Comment (pavel-esir) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:649]:
```
  -h, --help            show this help message and exit

Framework-agnostic parameters:
  --output_model OUTPUT_MODEL
                        This parameter is used to name output .xml/.bin files with converted model.
  --compress_to_fp16    Compress weights in output IR .xml/bin files to FP16.
```

probably we need it to be informative, state explicitly that user can specify True/False flags and that by default it is True

Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:649]: --compress_to_fp16 COMPRESS_TO_FP16
                        Compress weights in output IR .xml/bin files to FP16.

**Comment by slyalin** [comment]
Please point me to the place where we are preventing `compress_to_fp16=True` by default for Python API `ovc.convert_model` call.

**Comment by pavel-esir** [comment]
> Please point me to the place where we are preventing `compress_to_fp16=True` by default for Python API `ovc.convert_model` call.

ovc.convert_model does not even have such argument

https://github.com/openvinotoolkit/openvino/blob/9e8e77194187859f83026a5bcf7c0f4e6b22c875/tools/ovc/openvino/tools/ovc/convert.py#L19-L35

through `_convert` we get into `mock_emit_ir.py` which now does not call `compress_model(ov::Model)` at all. 
https://github.com/openvinotoolkit/openvino/pull/18652/files#diff-a52d3dad492c2819b72ce51c4eeb4a1b710ffb56620d77b55d71106faaeffd77

Therefore ovc.convert_model not only does not compress_to_fp16 by default but can not do that in principle even if user wanted.

**Comment by slyalin** [comment]
As we merged `save_model` which is doing compression by default, I think we can avoid changing `compress_to_fp16` parameter for `mo` in Python API, still changing it in `mo` as CLI tool. Benefits are: (1) this is alignment with `openvino.convert_model` that would give less surprises when people will migrate from `mo.convert_model` to `openvino.convert_model`, (2) less impact on "legacy" flow that consist of `mo.convert_model` and `serialize` and clear separation of it from a modern approach with `openvino.convert_model` + `openvino.save_model`.

File: tools/ovc/openvino/tools/ovc/cli_parser.py (hunk 3 above)
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:649]: Yes, please check this PR:
https://github.com/openvinotoolkit/openvino/pull/18785

**Review by slyalin — APPROVED** [review]

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_complex_params.py [hunk 4]
```diff
@@ -271,13 +280,13 @@ def test_mo_convert_tf_model_no_concat(self, params, ie_device, precision, ir_ve
...
-        {'params_test': {'input_shape': [Dimension(3, 10), 10, -1], 'use_convert_model_from_mo': True},
+        {'params_test': {'input_shape': [Dimension(3, 10), 10, -1], 'use_convert_model_from_mo': True, 'compress_to_fp16': True},
          'params_ref': {'input_shape': '[3..10,10,?]'}},
         {'params_test': {'input': [Dimension(3, 10), 10, -1]},
          'params_ref': {'input': 'Input[3..10,10,?]'}},
-        {'params_test': {'input': PartialShape([1, 100, 100, 3]), 'mean_values': [0.5, 1.3, 0.67], 'use_convert_model_from_mo': True},
+        {'params_test': {'input': PartialShape([1, 100, 100, 3]), 'mean_values': [0.5, 1.3, 0.67], 'use_convert_model_from_mo': True, 'compress_to_fp16': True},
          'params_ref': {'input': "Input[1,100,100,3]", 'mean_values': "[0.5,1.3,0.67]"}},
-        {'params_test': {'input': [1, 100, 100, 3], 'scale_values': [0.5, 1.3, 0.67], 'use_convert_model_from_mo': True},
+        {'params_test': {'input': [1, 100, 100, 3], 'scale_values': [0.5, 1.3, 0.67], 'use_convert_model_from_mo': True, 'compress_to_fp16': True},
```
[the remark this answers is not in the document]
Comment (slyalin) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_complex_params.py:315]: Doesn't look to be done exactly as discussed. But new tests are good.

**Comment by pavel-esir** [comment]
tests should be adjusted to the new behavior, ticket to that https://jira.devtools.intel.com/browse/CVS-116556

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_complex_params.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pavel-esir) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_complex_params.py:315]: ticket to adjust these tests to the new behavior https://jira.devtools.intel.com/browse/CVS-116556

**Pull request merged**